### PR TITLE
Merge omrport shared memory routines from OpenJ9

### PR DESCRIPTION
**Pull request by mthom**
This commit is in support of porting the refactored OpenJ9 shared cache into OMR.

Issue: #4610

Signed-off-by: Mark Thom <mark.thom@unb.ca>

### CODE REVIEW

**Comment by DmonteDD** [comment]
Dear reviewers,

This PR introduces shared memory routines from OpenJ9 to OMR. A closely associated PR #4758 , which introduced semaphore routines from OpenJ9 to OMR got merged recently. Hence, @youngar 's and @rwy0717 's comments and suggestions from PR #4758 are now incorporated in this PR too. I request you to please review.

Thank You!

**Comment by youngar** [comment]
@genie-omr build all

**Comment by DmonteDD** [comment]
It seems from the logs that the failure is due to undeclared identifiers and function prototypes. I don't know why it wasn't caught in other builds. I will cross-check and fix it up. Thanks.

**Comment by DmonteDD** [comment]
@youngar can u please build all? 
Thanks

**Comment by youngar** [comment]
@genie-omr build all

File: port/common/omrport.tdf [hunk 1]
```diff
@@ -673,6 +673,269 @@ TraceExit=Trc_PRT_sl_open_shared_library_Exit3 Group=sl Overhead=1 Level=10 NoEn
 
 TraceEvent=Trc_PRT_sl_open_shared_library_Event2 Group=sl Overhead=1 Level=10 NoEnv Template="omrsl_open_shared_library OS error message: %s"
 
+// omrshmem trace points
+TraceEntry=Trc_PRT_shmem_omrshmem_createFile_Entry Obsolete Group=omrshmem Overhead=1 Level=1 NoEnv Template="omrshmem createFile: Entry"
+TraceExit=Trc_PRT_shmem_omrshmem_createFile_Exit1 Obsolete Group=omrshmem Overhead=1 Level=1 NoEnv Template="omrshmem createFile: base file already exists!"
+TraceExit=Trc_PRT_shmem_omrshmem_createFile_Exit2 Obsolete Group=omrshmem Overhead=1 Level=1 NoEnv Template="omrshmem createFile: base file creation error: errno = %s"
+TraceExit=Trc_PRT_shmem_omrshmem_createFile_Exit Obsolete Group=omrshmem Overhead=1 Level=1 NoEnv Template="omrshmem createFile: Normal Exit"
```
Comment (keithc-ca) [inline port/common/omrport.tdf:680]: New tracepoints should added at the end of the file: fix #5523.

Comment (DmonteDD) [reply port/common/omrport.tdf:680]: Noted. Thank You @keithc-ca .

**Pull request merged**